spec/helpers/mock_project.rb

```diff
@@ -92,8 +92,18 @@ def bundle_install(version: nil)
       opts = {}
       opts[:chdir] = path
       Bundler.with_unbundled_env do
-        ::Gem.install("bundler", bundler_version)
-        out, err, status = Open3.capture3(["bundle", "_#{bundler_version}_", "install"].join(" "), opts)
+        cmd =
+          # prerelease versions are not always available on rubygems.org
+          # so in this case, we install whichever is the latest
+          if ::Gem::Version.new(bundler_version).prerelease?
```

Comment:
I tried that at the beginning, but the problem is:

1. Because we can't effectively know what's the latest available version, we can only return `nil` there.
2. When we get `nil` here, we need to change the command as well. So we still need an if condition like the current implementation.
3. When seeing a `nil` here, we also need to check `bundler_version` to see the reason. So we'll have this knowledge scattered at 2 places.